korge/src/commonMain/kotlin/com/soywiz/korge/input/SwipeComponent.kt

```diff
@@ -0,0 +1,139 @@
+package com.soywiz.korge.input
+
+import com.soywiz.korge.view.*
+import com.soywiz.korma.geom.*
+import kotlin.math.*
+
+data class SwipeInfo(
+    var dx: Double = 0.0,
+    var dy: Double = 0.0,
+    var direction: SwipeDirection
+)
+
+enum class SwipeDirection {
+    LEFT, RIGHT, TOP, BOTTOM
+}
+
+/**
+ * This methods lets you specify a [callback] to execute when swipe event is triggered.
+ * @property threshold a threshold for dx or dy after which a swipe event should be triggered.
+ * Negative [threshold] means a swipe event should be triggered only after [onUp] mouse event.
+ * [threshold] is set to -1 by default.
+ * @property direction a direction for which a swipe event should be triggered.
+ * [direction] is set to null by default, which means a swipe event should be triggered for any direction.
+ * @property callback a callback to execute
+ */
+fun <T : View> T.onSwipe(
+    threshold: kotlin.Number = -1,
+    direction: SwipeDirection? = null,
+    callback: suspend Views.(SwipeInfo) -> Unit
+): T {
+    var register = false
+    var sx = 0.0
+    var sy = 0.0
+    var cx = 0.0
+    var cy = 0.0
+    var movedLeft = false
+    var movedRight = false
+    var movedTop = false
+    var movedBottom = false
+    val thr = threshold.toDouble()
+    val view = this
+    val mousePos = Point()
+
+    fun views() = view.stage!!.views
+
+    fun updateMouse() {
+        val views = views()
+        mousePos.setTo(
+            views.nativeMouseX,
+            views.nativeMouseY
+        )
+    }
+
+    fun updateCoordinates() {
+        if (mousePos.x < cx) movedLeft = true
+        if (mousePos.x > cx) movedRight = true
+        if (mousePos.y < cy) movedTop = true
+        if (mousePos.y > cy) movedBottom = true
+        cx = mousePos.x
+        cy = mousePos.y
+    }
+
+    suspend fun triggerEvent(direction: SwipeDirection) {
+        register = false
+        views().callback(SwipeInfo(cx - sx, cy - sy, direction))
```

Comment:
To be consistent you should construct a dummy SwipeInfo instance and store it locally, then update it and send that same instance.
It is done like that to avoid allocations when using events.